js/util/window.js

```diff
@@ -30,12 +30,19 @@ function restore() {
     window.HTMLElement.prototype.clientLeft = 0;
     window.HTMLElement.prototype.clientTop = 0;
 
-    window.HTMLCanvasElement.prototype.getContext = function(type, attributes) {
-        if (!this._webGLContext) {
-            this._webGLContext = gl(this.width, this.height, attributes);
-        }
-        return this._webGLContext;
-    };
+    window.HTMLCanvasElement.prototype.getContext = (function () {
+        let original = window.HTMLCanvasElement.prototype.getContext;
```

Comment:
I don't think you need an self-invoking function here. You can just put the reference to original method outside of the scope and call it `originalGetContext`.